### DISCUSSION TITLE
How to add COM implementation at runtime by an addon?

### DISCUSSION THREAD

**Discussion by school510587** [nvaccess/nvda#16750] [General]
Hi developers,

Now NVDA manipulates UIAutomation COM objects via comtypes module. I'm developing one addon, which needs TSF functionality, i.e. ITfInputProcessorProfileMgr and ITfInputProcessorProfiles interfaces. I think the same skill can be applied here, but I don't find any helpful information from the developer guide. @josephsl has suggested me to start a new issue for discussion. Could anyone tell me how to add support for the two COM interfaces into NVDA at runtime?

Any response will be appreciated.

**Comment by Adriani90** [comment]
cc: @feerrenrut, @leonardder, @michaelDCurran
